lec_8_HW/task_2.js

```diff
@@ -0,0 +1,179 @@
+const enterprises = [
+    {
+        id: 1,
+        name: "Предприятие 1",
+        departments: [
+            {
+                id: 2,
+                name: "Отдел тестирования",
+                employees_count: 10,
+            },
+            {
+                id: 3,
+                name: "Отдел маркетинга",
+                employees_count: 20,
+            },
+            {
+                id: 4,
+                name: "Администрация",
+                employees_count: 15,
+            },
+        ],
+    },
+    {
+        id: 5,
+        name: "Предприятие 2",
+        departments: [
+            {
+                id: 6,
+                name: "Отдел разработки",
+                employees_count: 50,
+            },
+            {
+                id: 7,
+                name: "Отдел маркетинга",
+                employees_count: 20,
+            },
+            {
+                id: 8,
+                name: "Отдел охраны труда",
+                employees_count: 5,
+            },
+        ],
+    },
+    {
+        id: 9,
+        name: "Предприятие 3",
+        departments: [
+            {
+                id: 10,
+                name: "Отдел аналитики",
+                employees_count: 0,
+            },
+        ],
+    },
+];
+
+// 1. ///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+function printEnterprisesWithDepartments() {
+    enterprises.forEach(enterprise => {
+        const totalEmployees = enterprise.departments.reduce((sum, department) => sum + department.employees_count, 0);
+        console.log(`${enterprise.name} (${totalEmployees || "нет"} сотрудников)`);
+        enterprise.departments.forEach(department => {
+            console.log(`- ${department.name} (${department.employees_count || "нет"} сотрудников)`);
+        });
+    });
+}
+
+// printEnterprisesWithDepartments();
+
+// 2. ///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+function getEnterpriseName(identifier) {
+    for (const enterprise of enterprises) {
+        for (const department of enterprise.departments) {
+            if (department.id === identifier || department.name === identifier) {
+                return enterprise.name;
+            }
+        }
+    }
+    return null;
+}
+
+// console.log(getEnterpriseName(4));
+// console.log(getEnterpriseName("Отдел маркетинга"));
+
+// 3. ///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+function addEnterprise(name) {
+    const newId = enterprises.reduce((maxId, enterprise) => Math.max(maxId, enterprise.id), 0) + 1;
```

Comment:
need to create function for new id creation among all enterprises and departments